Orleans.Sagas/SagaGrain.cs

```diff
@@ -22,8 +22,35 @@ public SagaGrain(IGrainActivationContext grainContext, IServiceProvider serviceP
             this.grainContext = grainContext;
             this.serviceProvider = serviceProvider;
             this.logger = logger;
-        }
-
+        }
+
+        protected override async Task ReadStateAsync()
+        {
+            try
+            {
+                await base.ReadStateAsync();
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(1, ex, "Failed to read state");
+                await HandleReadingStateError();
+            }
+        }
+
+        private async Task HandleReadingStateError()
+        {
+            try
+            {
+                await UnRegisterReminderAsync();
+                isActive = false;
+                await ClearStateAsync();

```

Comment:
Would this not clear a saga's state in the event of any exception? For example an intermittent storage access problem?

Reply:
Good catch, so maybe to need to catch just specific exceptions like InvalidCastException.

@creyke WDYT ?

Reply:
In addition if there is serialization error, retry wont fix the exception, and will be in infinite loop of activating the activity, so I prefer in this case to clear the state ( unreachable data)

Reply:
Agreed. Catch only exceptions which are non-recoverable.

Reply:
Are you aware of other types of exception that need to be watched ( in addition to InvalidCastException)